Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -2291,24 +2291,27 @@ static bool isFlexItem(const RenderObject* renderer)
     return false;
 }
 
-static bool rendererContainingBlockHasMarginTrim(const RenderBox& renderer, std::optional<MarginTrimType> marginTrimType)
+static bool rendererCanHaveMarginTrimRareDataSetWithinFormattingContext(const RenderBox& renderer, std::optional<MarginTrimType> marginTrimType)
 {
+    // A renderer will have a specific margin marked as trimmed by setting its rare data bit if:
+    // 1.) The layout system the box is in has this logic (setting the rare data bit for this 
+    // specific margin) implemented
+    // 2.) The block container/flexbox/grid has this margin specified in its margin-trim style
+    // If marginTrimType is empty we will check if any of the supported margins are in the style
     auto* containingBlock = renderer.containingBlock();
     if (!containingBlock || containingBlock->isRenderView())
         return false;
```

Comment:
Any reason for checking containing block even before establishing that we are dealing with BFC here?

Reply:
I believe this was related to some test cases there were failing with a null containing block (so I added it after seeing the test results). The logic here should probably be rearranged and should attempt to first figure out the formatting context before attempting a check like this. I can do this in the same patch when I clean up the incorrect uses of `containingBlock` (unless you would rather have it fixed here)